src/main/cpp/core/Cluster.h

```diff
@@ -36,17 +36,16 @@ namespace stride {
 class RngHandler;
 class Calendar;
 
+using ClusterID = std::size_t;
```

Comment:
Maybe you should rename this to `ClusterId` for consistency with `PersonId` and `RegionId`. I've noticed that `CityID` and `ProvinceID` also use the all--upper-case suffix. Alternatively, you can rename `PersonId` and `RegionId` to match your `*ID` naming convention.